bhv/symbolic.py

```diff
@@ -617,6 +617,44 @@ def expected_active_fraction(self, **kwargs):
         return 1. - PoiBin([v.expected_active_fraction(**kwargs) for v in self.vs]).cdf(len(self.vs)//2)
 
 
+@dataclass
+class Representative(SymbolicBHV):
+    vs: list[SymbolicBHV]
+
+    def labeled_children(self, **kwargs):
+        return list(zip(self.vs, map(str, range(len(self.vs)))))
+
+    def reconstruct(self, *cs):
+        return Representative(list(cs))
+
+    def show(self, **kwargs):
+        args = format_list((v.show(**kwargs) for v in self.vs), **{k: kwargs[k] for k in ["indent", "aindent", "newline_threshold"] if k in kwargs})
+        return kwargs.get("impl", "") + f"representative({args})"
+
+    def instantiate(self, **kwargs):
+        return kwargs.get("bhv").representative([v.execute(**kwargs) for v in self.vs])
+
+    # def expected_active_fraction(self, **kwargs):
+    #     from .poibin import PoiBin
+    #    return 1. - PoiBin([v.expected_active_fraction(**kwargs) for v in self.vs]).cdf(len(self.vs)//2)
+
+    def expected_error(self, x: Var):
+        """
+        Gives expected bit error rate between an expression that contains only representative operators and random
+        hypervectors, and a given random hypervector.
+        """
+        return self._expected_error(self, x)
+
+    def _expected_error(self, r, x):
+        if isinstance(r, Var):
```

Comment:
This code should not just work for checking the existence of Vars but also of a specific random vector that's shared. E.g., in `v = rand(); r = R(Var('a'), v, v)`, I still want to know that `v` is represented more than the Var.

Reply:
Possibly, this is fine, and we just want a "rand_to_var" method that treats random hypervectors (or certain sub-expressions) as vars.

Reply:
Yes, I assumed that all leaves would be of type Var, but we can add checks.

Reply:
If v=random(a) with a an active ratio != 1/2 we would also need to take this into account. Currently I assume that all Vars are random(1/2).

Reply:
Ahh we don't know what active fractions vars are going to be: they can be ZERO, too.
If you looked at "expected active fraction", we want to handle it the same way.